Description: Disable tests for integer overflowing for 32-bit architectures.
 Disables some tests for overflowing INTEGER and OID types for ILP32 machines.
 Tests are fair but specified behaviour is not implemented yet.
Forwarded: not-needed
Author: Eugene Seliverstov <theirix@gmail.com>
Last-Update: 2012-10-01
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/asn1c/tests/check-assembly.sh
+++ b/asn1c/tests/check-assembly.sh
@@ -39,7 +39,7 @@
 cat > Makefile <<EOM
 # This file is autogenerated by ../$0
 
-COMMON_FLAGS= -I. -DEMIT_ASN_DEBUG=0
+COMMON_FLAGS= -I. -DEMIT_ASN_DEBUG=0 -DNO_OVERFLOW_TESTS
 CFLAGS = \${COMMON_FLAGS} ${CFLAGS} -g -O0
 CXXFLAGS = \${COMMON_FLAGS} ${CXXFLAGS}
 
--- a/skeletons/tests/check-INTEGER.c
+++ b/skeletons/tests/check-INTEGER.c
@@ -256,12 +256,14 @@
 		check_xer(-1, "<INTEGER>2147483649</INTEGER>", 0);
 		check_xer(-1, "<INTEGER>3147483649</INTEGER>", 0);
 		check_xer(-1, "<INTEGER>4147483649</INTEGER>", 0);
+#ifndef NO_OVERFLOW_TESTS
 		check_xer(-1, "<INTEGER>5147483649</INTEGER>", 0); /* special */
 		check_xer(-1, "<INTEGER>9147483649</INTEGER>", 0);
 		check_xer(-1, "<INTEGER>9999999999</INTEGER>", 0);
 		check_xer(-1, "<INTEGER>-5147483649</INTEGER>", 0);/* special */
 		check_xer(-1, "<INTEGER>-9147483649</INTEGER>", 0);
 		check_xer(-1, "<INTEGER>-9999999999</INTEGER>", 0);
+#endif
 	}
 	if(sizeof(long) == 8) {
 		check_xer(0, "<INTEGER>2147483648</INTEGER>", 2147483648);
--- a/skeletons/tests/check-OIDs.c
+++ b/skeletons/tests/check-OIDs.c
@@ -416,6 +416,7 @@
 	check_parse("1.2000000000.3", 3);
 	check_parse("1.2147483647.3", 3);
 	if(sizeof(long) == 4) {
+#ifndef NO_OVERFLOW_TESTS
 		check_parse("1.2147483648.3", -1);	/* overflow on ILP32 */
 		check_parse("1.2147483649.3", -1);	/* overflow on ILP32 */
 		check_parse("1.3000000000.3", -1);
@@ -423,6 +424,7 @@
 		check_parse("1.5000000000.3", -1);
 		check_parse("1.6000000000.3", -1);
 		check_parse("1.9000000000.3", -1);
+#endif
 	} else if(sizeof(long) == 8) {
 		check_parse("1.2147483648.3", 3);
 		check_parse("1.9223372036854775807.3", 3);
--- a/skeletons/tests/Makefile.am
+++ b/skeletons/tests/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I$(top_srcdir)/skeletons
+AM_CPPFLAGS = -I$(top_srcdir)/skeletons -DNO_OVERFLOW_TESTS
 
 check_PROGRAMS = \
 	check-ber_tlv_tag	\
